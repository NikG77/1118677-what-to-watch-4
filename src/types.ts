export interface FilmType {
  backgroundColor: string,
  director: string,
  description: string,
  duration: number,
  id: number,
  isFavorite: boolean,
  genre: string,
  pictureBackground: string,
  poster: string,
  previewVideo: string,
  ratingScore: number,
  releaseDate: number,
  ratingCount: number,
  src: string,
  starring:Array<string>,
  title: string,
  videoLink: string,
}


export interface UserInfoInterface {
  id: number;
  email: string;
  name: string;
  avatarUrl: string;
}


export interface Comment {
    author: string;
    date: number;
    id: string;
    idUser: number;
    rating: number;
    text: string;
}



