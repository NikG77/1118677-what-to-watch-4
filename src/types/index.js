import PropTypes from "prop-types";

export const filmsType = PropTypes.arrayOf(PropTypes.shape({
  title: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired,
  ratingScore: PropTypes.number.isRequired,
  director: PropTypes.string.isRequired,
  starring: PropTypes.arrayOf(PropTypes.string).isRequired,
  genre: PropTypes.string.isRequired,
  releaseDate: PropTypes.number.isRequired,
  pictureBackground: PropTypes.string.isRequired,
}));

export const mainFilmType = PropTypes.shape({
  genre: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  releaseDate: PropTypes.number.isRequired,
  poster: PropTypes.string.isRequired,
  pictureBackground: PropTypes.string.isRequired,
});

export const filmType = PropTypes.shape({
  title: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired,
  ratingScore: PropTypes.number.isRequired,
  director: PropTypes.string.isRequired,
  starring: PropTypes.arrayOf(PropTypes.string).isRequired,
  genre: PropTypes.string.isRequired,
  releaseDate: PropTypes.number.isRequired,
  pictureBackground: PropTypes.string.isRequired,
});


