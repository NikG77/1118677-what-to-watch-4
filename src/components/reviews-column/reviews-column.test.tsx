import * as React from "react";
import * as renderer from "react-test-renderer";
import ReviewsColumn from "./reviews-column";
import {Comment} from "../../types";

const reviewsColumns: Comment[] = [
  {
    author: `Quentin Tarantino`,
    date: 1677777,
    id: `1111`,
    idUser: 15,
    rating: 9,
    text: `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
  },
];

it(`Should ReviewsColumn render correctly`, () => {
  const tree = renderer
    .create(<ReviewsColumn
      reviewsColumns={reviewsColumns}
    />).toJSON();

  expect(tree).toMatchSnapshot();
});
