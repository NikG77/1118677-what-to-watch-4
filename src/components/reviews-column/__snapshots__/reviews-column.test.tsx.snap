// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ReviewsColumn render correctly 1`] = `
<div
  className="movie-card__reviews-col"
>
  <div
    className="review"
  >
    <blockquote
      className="review__quote"
    >
      <p
        className="review__text"
      >
        Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.
      </p>
      <footer
        className="review__details"
      >
        <cite
          className="review__author"
        >
          Quentin Tarantino
        </cite>
        <time
          className="review__date"
          dateTime="1970-01-01"
        >
          January 1, 1970
        </time>
      </footer>
    </blockquote>
    <div
      className="review__rating"
    >
      9
    </div>
  </div>
</div>
`;
